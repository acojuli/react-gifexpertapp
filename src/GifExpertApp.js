import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Pokemon'])

    // const handleAdd = () => {

    //     setCategories( [ ...categories, 'Ranma' ] );
    //     // Definimos un callback en el cual el primer argumento es el estado anterior y luego el nuevo estado que tieen que regresar.
    //     // setCategories( cats => [ ...cats, 'Ranma '] );
    // }



    return (
        <>
            <h2>GIFs Search Engine</h2>
            <p>Place the text you want in the search box and press enter. Then the app will show related GIFs.</p>
            <AddCategory setCategories={ setCategories }/>
            <hr />

            <ol>
                { 
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category } />

                    ))
                }
            </ol>
        </>
    );
    };

export default GifExpertApp;