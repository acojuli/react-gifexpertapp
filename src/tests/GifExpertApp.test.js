import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas del componente <GifExpertApp />', () => {

    test('should show the component correctly', () => {

        const wrapper = shallow( <GifExpertApp /> )
        expect( wrapper ).toMatchSnapshot();
        
    });    

    test('should show a category list', () => {

        const categories = ['Pokemon', 'Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        // expect( wrapper.find('GifGrid').lenth ).toBe( categories.length );
        
    })
    
    
})
