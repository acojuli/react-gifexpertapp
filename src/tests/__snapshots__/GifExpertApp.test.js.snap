// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifExpertApp /> should show a category list 1`] = `
<Fragment>
  <h2>
    GIFs Search Engine
  </h2>
  <p>
    Place the text you want in the search box and press enter. Then the app will show related GIFs.
  </p>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Pokemon"
      key="Pokemon"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas del componente <GifExpertApp /> should show the component correctly 1`] = `
<Fragment>
  <h2>
    GIFs Search Engine
  </h2>
  <p>
    Place the text you want in the search box and press enter. Then the app will show related GIFs.
  </p>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
